### PR TITLE
[BYOC-DNNL] add post_sum pattern

### PR DESCRIPTION
**Pull request by crazydemo**
This PR add `conv2d-add-sum-relu` pattern, and the corresponding test case is added.

### CODE REVIEW

File: tests/python/contrib/test_dnnl.py [hunk 1]
```diff
@@ -192,6 +192,7 @@ def run_and_verify(mod, input, params, target, run_module, subgraph_num=None, te
             if use_dnnl:
                 processed_mod = partition_for_dnnl(processed_mod, params, alter_layout)
                 check_dnnl_used(processed_mod)
+                print(processed_mod)
```
Comment (billishyahao) [inline tests/python/contrib/test_dnnl.py:195]: remove it

File: python/tvm/relay/op/contrib/dnnl.py [hunk 2]
```diff
@@ -305,6 +395,12 @@ def pattern_table():
     dnnl_patterns = list()
     dnnl_patterns.append(make_qnn_conv2d_pattern())
     dnnl_patterns.append(make_qnn_dense_pattern())
+    dnnl_patterns.append(
+        ("dnnl.conv2d_bias_sum_relu", make_conv_bias_sum_relu_pattern("nn.conv2d"), make_predicate(add_checker))
+    ),
+    dnnl_patterns.append(
+        ("dnnl.conv2d_bias_sum", make_conv_bias_sum_relu_pattern("nn.conv2d", False), make_predicate(add_checker))
+    ),
```
Comment (yangulei) [inline python/tvm/relay/op/contrib/dnnl.py:403]:
```suggestion
    dnnl_patterns.append(
        (
            "dnnl.conv2d_bias_sum_relu",
            make_conv_bias_sum_relu_pattern("nn.conv2d"),
            make_predicate(add_checker),
        )
    )
    dnnl_patterns.append(
        (
            "dnnl.conv2d_bias_sum",
            make_conv_bias_sum_relu_pattern("nn.conv2d", False),
            make_predicate(add_checker),
        )
    )
```

File: src/runtime/contrib/dnnl/dnnl_json_runtime.cc [hunk 3]
```diff
@@ -361,6 +367,10 @@ class DNNLJSONRuntime : public JSONRuntimeBase {
 
     // TODO(@apeskov): Simulation of inplace primitive. just as PoC.
     auto sum_in_tr = GetInputByName(nid, "sum_idx").TreatAs(dst_layout);
+    if (op_name.find("_sum") != std::string::npos) {
+      sum_in_tr = GetInput(nid, node.GetInputs().size()-1);
```
Comment (yangulei) [inline src/runtime/contrib/dnnl/dnnl_json_runtime.cc:371]:
```suggestion
      sum_in_tr = GetInput(nid, node.GetInputs().size() - 1);
```

File: tests/python/contrib/test_dnnl.py [hunk 4]
```diff
@@ -788,6 +789,44 @@ def test_conv2d_pattern(run_module, dtype="float32"):
...
+            scale=True,
+            epsilon=1e-5,
+        )
+        sum_data = relay.var("data1", shape=sum_shape, dtype=dtype)
+        out = relay.add(out, sum_data)
+        dic["data1"] = sum_shape
+        param_lst += ["data1"]
+        return relay.nn.relu(out), dic, param_lst
+
+    conv2d_bn_sum_relu, dic, param_lst = get_conv2d_bn_sum_relu(x_shape, k_shape, sum_shape=(1, 16, 6, 6), dtype=dtype)
```
Comment (yangulei) [inline tests/python/contrib/test_dnnl.py:819]:
```suggestion
    conv2d_bn_sum_relu, dic, param_lst = get_conv2d_bn_sum_relu(
        x_shape, k_shape, sum_shape=(1, 16, 6, 6), dtype=dtype
    )
```

File: tests/python/contrib/test_dnnl.py [hunk 5]
```diff
@@ -788,6 +789,44 @@ def test_conv2d_pattern(run_module, dtype="float32"):
...
+    conv2d_bn_sum_relu = tvm.IRModule.from_expr(conv2d_bn_sum_relu)
+    config = conv2d_bn_sum_relu, dic, param_lst
+    run_and_verify_func(config, run_module=run_module, dtype=dtype)
+
+    conv2d_bn_sum_relu, dic, param_lst = get_conv2d_bn_sum_relu(x_shape, k_shape, sum_shape=(1, 16, 1, 1), dtype=dtype)
```
Comment (yangulei) [inline tests/python/contrib/test_dnnl.py:824]:
```suggestion
    conv2d_bn_sum_relu, dic, param_lst = get_conv2d_bn_sum_relu(
        x_shape, k_shape, sum_shape=(1, 16, 1, 1), dtype=dtype
    )
```

File: tests/python/contrib/test_dnnl.py [hunk 6]
```diff
@@ -1732,4 +1771,5 @@ def generate_model(p, c):
 
 
 if __name__ == "__main__":
-    tvm.testing.main()
+    # tvm.testing.main()
+    test_conv2d_bias_sum_relu(True)
```
Comment (yangulei) [inline tests/python/contrib/test_dnnl.py:1775]:
```suggestion
    tvm.testing.main()

```

File: tests/python/contrib/test_dnnl.py (hunk 1 above)
Comment (crazydemo) [reply tests/python/contrib/test_dnnl.py]: Fixed.

File: python/tvm/relay/op/contrib/dnnl.py (hunk 2 above)
Comment (crazydemo) [reply python/tvm/relay/op/contrib/dnnl.py]: Fixed.

File: src/runtime/contrib/dnnl/dnnl_json_runtime.cc (hunk 3 above)
Comment (crazydemo) [reply src/runtime/contrib/dnnl/dnnl_json_runtime.cc]: Fixed.

File: tests/python/contrib/test_dnnl.py (hunk 4 above)
Comment (crazydemo) [reply tests/python/contrib/test_dnnl.py]: Fixed.

File: tests/python/contrib/test_dnnl.py (hunk 5 above)
Comment (crazydemo) [reply tests/python/contrib/test_dnnl.py]: Fixed.

File: tests/python/contrib/test_dnnl.py (hunk 6 above)
Comment (crazydemo) [reply tests/python/contrib/test_dnnl.py]: Fixed.

**Comment by crazydemo** [comment]
@masahi Could you please review this PR? This PR adds `conv2d-add-sum-relu pattern` with required checks, and the corresponding test case is added.

**Review by masahi — APPROVED** [review]

**Pull request merged**